internal/services/runnerService.go

```diff
@@ -131,30 +127,54 @@ func (r *runnerService) RunSolution(task *TaskForRunner, messageID string) s.Sol
 			UseChroot:     task.useChroot,
 		}
 
-		execResult := exec.ExecuteCommand(filePath, messageID, commandConfig)
+		execResult := solutionExecutor.ExecuteCommand(filePath, messageID, commandConfig)
 		switch execResult.ExitCode {
 		case constants.ExitCodeSuccess:
 			r.logger.Infof("Comparing output %s with expected output [MsgID: %s]", outputPath, messageID)
 			expectedFilePath := fmt.Sprintf("%s/%s/%d.out", task.taskFilesDirPath, task.outputDirName, (i + 1))
-			result, difference, err := verifier.CompareOutput(outputPath, expectedFilePath)
+
+			diffCmd := exec.Command("diff", "-w", expectedFilePath, outputPath)
+			diffCmdOutput, err := os.OpenFile(stderrPath, os.O_CREATE|os.O_WRONLY, 0644)
 			if err != nil {
-				r.logger.Errorf("Error comparing output %s with expected output [MsgID: %s]: %s", outputPath, messageID, err.Error())
-				solutionSuccess = false
-				difference = err.Error()
-			}
-			if !result {
-				solutionSuccess = false
+				r.logger.Errorf("Error opening output file %s [MsgID: %s]: %s", outputPath, messageID, err.Error())
+				testCases[i] = s.TestResult{
+					Passed:       false,
+					ErrorMessage: err.Error(),
+					Order:        (i + 1),
+				}
+				solutionStatus = s.InternalError
+				solutionMessage = constants.SolutionMessageInternalError
+				continue
 			}
-			err = StoreTestCaseDifferenceInErrFile((i + 1), difference, commandConfig.StderrPath)
+			defer diffCmdOutput.Close()
+
+			diffCmd.Stdout = diffCmdOutput
+			err = diffCmd.Run()
+			diffExitCode := diffCmd.ProcessState.ExitCode()
```

Comment:
I would preffer this to be wrapped in verifier interface still. This way in future if we would need our own verifier, which we plan to have. It will be easier to replace